src/singlelayerqg.jl

```diff
@@ -106,7 +106,13 @@ function Problem(dev::Device=CPU();
   # topographic PV
   eta === nothing && (eta = zeros(dev, T, (nx, ny)))
 
-  params = deformation_radius == Inf ? BarotropicQGParams(grid, T(β), T(U), eta, T(μ), T(ν), nν, calcF) : EquivalentBarotropicQGParams(grid, T(β), T(deformation_radius), T(U), eta, T(μ), T(ν), nν, calcF)
+  if U isa Number
+	U = T(U)
+  else
+	U = device_array(dev)(reshape(U,(1, grid.ny)))
```

Comment:
```suggestion
	U = reshape(U, (1, grid.ny)))
	U = device_array(dev, U)
```

might be wrong but I think this is how it's supposed to be used?

also it doesn't seem necessary to require that `U` is an array (it could be a function too right? then it can change in time). But I think it's ok for this PR just to add this and make a comment that supporting additional types is future work

Reply:
```
help?> device_array
search: device_array

  device_array(device::Device)
  device_array(device::Device, T, dim)

  Return the proper array type according to the device, i.e., Array for CPU and CuArray for GPU.

  ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  device_array(grid::AbstractGrid)

  Return the proper array type according to the grid's device, i.e., Array for CPU and CuArray for GPU.
```